packages/compass-workspaces/src/provider.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useContext, useRef } from 'react';
+import { useSelector, useStore } from './stores/context';
+import type {
+  OpenWorkspaceOptions,
+  TabOptions,
+  WorkspaceTab,
+} from './stores/workspaces';
+import {
+  getActiveTab,
+  openWorkspace as openWorkspaceAction,
+} from './stores/workspaces';
+
+function useWorkspacesStore() {
+  try {
+    return useStore();
+  } catch {
+    throw new Error(
+      "Can't find Workspaces store in React context. Make sure you are using workspaces service and hooks inside Workspaces scope"
+    );
```

Comment:
It's safe to assume that the original error message will always be less helpful than this and there's no need to include it in the wrapped error?

Reply:
I'm pretty sure, yeah. This `useStore` method is a [very thin wrapper](https://github.com/reduxjs/react-redux/blob/8c503efef000b69133accaecfec8fff346822a3e/src/hooks/useReduxContext.ts#L12-L24) over `useContext` with just a special error message provided